src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1130,6 +1116,14 @@ private void joinTokenRing(int delay, boolean autoBootstrap, Collection<String>
         }
     }
 
+    private static boolean isSchemaConsistent()
+    {
+        String localSchemaVersion = Schema.instance.getVersion().toString();
+        return Gossiper.instance.getEndpointStates().stream()
+                                .map(entry -> entry.getValue().getApplicationState(ApplicationState.SCHEMA).value)
+                                .allMatch(localSchemaVersion::equals);
```

Comment:
ah wait doesn't this check cause problems bc the endpoint states can have `LEFT` nodes?